pkg/mysql/topology.go

```diff
@@ -0,0 +1,52 @@
+package mysql
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/replicator"
+)
+
+func GetTopology(hosts []string, operatorPass string) (string, []string, error) {
+	replicas := sets.NewString()
+	primary := ""
+
+	for _, host := range hosts {
+		p, err := getPrimary(host, operatorPass)
+		if err != nil {
+			return "", nil, errors.Wrap(err, "get primary")
+		}
+		if p != "" {
+			primary = p
+		}
+		replicas.Insert(host)
+	}
+	if primary == "" && len(hosts) == 1 {
+		primary = hosts[0]
+	} else if primary == "" {
+		primary = replicas.List()[0]
+	}
+	if replicas.Len() > 0 {
+		replicas.Delete(primary)
+	}
+	return primary, replicas.List(), nil
+}
+
+func getPrimary(host, operatorPass string) (string, error) {
```

Comment:
@pooknull Are you sure that  this check will work in case of external replication to our PRIMARY ?

Reply:
Also, why not use `orchestrator.ClusterPrimary(..)` function?